codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeCodegenMetadata.java

```diff
@@ -33,44 +34,38 @@ public final class AuthSchemeCodegenMetadata {
-                                                .valueProvider(utils -> "params.region().id()")
-                                                .valueType(ValueType.EXPRESSION)
+                                                .valueEmitter((spec, utils) -> spec.addCode("$L", "params.region().id()"))
                                                 .build())
         .build();
 
-    static final AuthSchemeCodegenMetadata S3 = builder()
-        .schemeId(AwsV4AuthScheme.SCHEME_ID)
-        .addProperty(SignerPropertyValueProvider.builder()
-                                                .containingClass(AwsV4HttpSigner.class)
-                                                .fieldName("SERVICE_SIGNING_NAME")
-                                                .valueProvider(AuthSchemeSpecUtils::signingName)
-                                                .valueType(ValueType.STRING)
-                                                .build())
-        .addProperty(SignerPropertyValueProvider.builder()
-                                                .containingClass(AwsV4HttpSigner.class)
-                                                .fieldName("REGION_NAME")
-                                                .valueProvider(utils -> "params.region().id()")
-                                                .valueType(ValueType.EXPRESSION)
-                                                .build())
-        .addProperty(SignerPropertyValueProvider.builder()
-                                                .containingClass(AwsV4HttpSigner.class)
-                                                .fieldName("DOUBLE_URL_ENCODE")
-                                                .valueProvider(utils -> "false")
-                                                .valueType(ValueType.EXPRESSION)
-                                                .build())
-        .addProperty(SignerPropertyValueProvider.builder()
-                                                .containingClass(AwsV4HttpSigner.class)
-                                                .fieldName("NORMALIZE_PATH")
-                                                .valueProvider(utils -> "false")
-                                                .valueType(ValueType.EXPRESSION)
-                                                .build())
-        .build();
+    static final AuthSchemeCodegenMetadata SIGV4_UNSIGNED_BODY =
+        SIGV4.toBuilder()
+             .addProperty(SignerPropertyValueProvider.builder()
+                                                     .containingClass(AwsV4HttpSigner.class)
+                                                     .fieldName(
+                                                         "PAYLOAD_SIGNING_ENABLED")
+                                                     .valueEmitter((spec, utils) -> spec.addCode("$L", false))
+                                                     .build())
+             .build();
+
+    static final AuthSchemeCodegenMetadata S3 =
```

Comment:
S3 by default needs to actually have PAYLOAD_SIGNING_ENABLED=false. Based on https://github.com/aws/aws-sdk-java-v2/blob/4ea3cc28a0825fa101547fd82cfb2b5767b040d9/core/auth/src/main/java/software/amazon/awssdk/auth/signer/params/AwsS3V4SignerParams.java#L91. This is not model driven, but desired behavior of the S3 SDK.

Reply:
Note this is only for `s3`, `s3control` (which maps to authType=s3v4) is PAYLOAD_SIGNING_ENABLED=true (the default, like all Sigv4 services), based on https://github.com/aws/aws-sdk-java-v2/blob/4ea3cc28a0825fa101547fd82cfb2b5767b040d9/services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/PayloadSigningInterceptor.java#L36

Reply:
Note, the s3 model has v4-unsigned-body explicitly in the model for `WriteGetObjectResponse`, which, I'm not sure why, but may mean that this operation _has_ to use unsigned payload, while for other S3 operations, both signed and unsigned work and we just want the SDK to default to unsigned.

Reply:
> S3 by default needs to actually have PAYLOAD_SIGNING_ENABLED=false. Based on
> 
> https://github.com/aws/aws-sdk-java-v2/blob/4ea3cc28a0825fa101547fd82cfb2b5767b040d9/core/auth/src/main/java/software/amazon/awssdk/auth/signer/params/AwsS3V4SignerParams.java#L91
> 
> . This is not model driven, but desired behavior of the S3 SDK.

Looks like the logic is more involved than that, the javadocs says

> Payload signing is optional when chunked encoding is not used and requests are made against an HTTPS endpoint.  Under these conditions the client will by default opt to not sign payloads to optimize performance.  If this flag is set to true the client will instead always sign payloads.

I don't think we should do that unconditionally but might be wrong about it.